### DISCUSSION TITLE
Speed up KMeans using FAISS

### DISCUSSION THREAD

**Discussion by lukegre** [ArcticSnow/TopoPyScale#111] [Ideas]
I know that clustering probably isn't the biggest bottleneck in the process, but I've used FAISS in the past, which is a lot faster. 
This provides a few more details: https://www.kdnuggets.com/2021/01/k-means-faster-lower-error-scikit-learn.html
This might make searching for the number of optimal clusters quite a bit quicker (20 x if the article is anything to go by).

**Comment by ArcticSnow** [comment]
Thank you @lukegre for the suggestion. This looks interesting indeed! As you said the biggest bottleneck isn't quite the clustering step. If you feel like it you are welcome to push a PR adding this method.

**Reply by lukegre** [reply]
Did a quick test, FAISS is much faster than regular Kmeans, but not than MiniBatch. So, probably not worth the effort.

**Reply by ArcticSnow** [reply]
Good. As you had pointed out, clustering is not the bottleneck. But thanks for suggesting and testing.
